app/src/lang/translations/en-US.yaml

```
@@ -773,6 +773,7 @@ flows: Flows
 flow: Flow
 select_a_flow: Select a Flow
 run_flow_on_current: Run flow on Current Item
+run_flow_on_current_edited_confirm: This item may be updated by this flow, are you sure you want to execute it?
```

```suggestion
run_flow_on_current_edited_confirm: This Flow might change fields you've already changed, but not saved yet. Are you sure you want to execute it?
```

I'm trying to find a slightly better phrasing here. @benhaynes thoughts?

---

TL;DR: When running the flow, it will update the item in the background. It won't cancel your made changes. However, that does mean that the flow might update a field value in the background of a field that you just changed, so you don't see the "updated" value in the background, as your change overrides it 🤔

---

Is the Flow update before or after the manual update? Does one always override the other... and if so, which?

---

The flow runs whenever the user runs it manually, so this can be before or after the user made changes. 

If the user made changes to a field that the flow also changed, the user's changes are not overridden.

---

OK, so just to confirm: If you change a value, and then manually run a flow, then the flow can/may update field values... but **not** any that were manually updated (even if not saved yet)?

If it may override unsaved changes, then I'm thinking something like this:
```
Running this Flow may override unsaved changes you've made to this form. Would you like to continue?
```

---

> OK, so just to confirm: If you change a value, and then manually run a flow, then the flow can/may update field values... but not any that were manually updated (even if not saved yet)?

The flow will update the stored item values in the background, and doesn't touch any of the user's edits. Just to reiterate, "edits" are always merged on top of the original item in the UX, so for example:

1. User opens page
2. App downloads current version of item. Lets say `title = example`
3. User changes `title = updated`
4. User runs flow
5. Flow refreshes "current" version of item
6. The user still sees `title = updated`, as that's the latest edit. However, the previously stored `title = example` is now `title = flow-changed-this`

So the user's edit is never overwritten, but they would also not be aware of the title being changed in the background, as their edit is overwriting it again 🤔

---

Quick video which shows exactly what Rijk described:

https://user-images.githubusercontent.com/42867097/182087952-0de1e56b-80c0-4aff-bdb1-b227a6aa0da6.mp4

---

@azrikahar Lets go with yours for now, as it's less confusing to read (even though it's _technically_ less accurate) 👍🏻